### ISSUE TITLE
Edlen Ring support

### ISSUE THREAD

**Issue by IgorGHK** [Quidrex/DSDeaths#10]
Hey, your program is great! Any options with Elden Ring?

**Comment by Quidrex** [comment]
I don't own Elden Ring yet and I'm not sure if I will buy it in the near future. If someone else finds the memory address I can of course add it but keep in mind that it will likely break with every patch.

**Comment by slyyyy78** [comment]
Can you describe how do you find correct addres? I assumed that you are using some memory scanner and the you you are changeing counter in game for example 101 deaths and tyhe you are scannig memory where those values are . Can you point good tool for doing this ?

**Comment by Danzoui** [comment]
> Can you describe how do you find correct addres? I assumed that you are using some memory scanner and the you you are changeing counter in game for example 101 deaths and tyhe you are scannig memory where those values are . Can you point good tool for doing this ?

Cheat Engine

**Comment by slyyyy78** [comment]
Unfortunately the game detects this software . I could create patch if someone could find the address.

**Comment by Danzoui** [comment]
Try this to bypass EAC:
[Elden.Ring.EAC.Bypass.zip](https://github.com/Quidrex/DSDeaths/files/8143643/Elden.Ring.EAC.Bypass.zip)

**Comment by slyyyy78** [comment]
I found: 7FF338797CB4

**Comment by slyyyy78** [comment]
I could do the patch, I'm java programmer but I don't think that the change is difficult. But how to build release ?

**Comment by Danzoui** [comment]
> I could do the patch, I'm java programmer but I don't think that the change is difficult. But how to build release ?

You can do it in the Visual Studio.

**Comment by slyyyy78** [comment]
I don't think it's correct address. Me second gave me different result. Please me correct if I'm wrong. Addres column is giving me relative address for a process of game or it's absolute address in my PC ? If it's absolute address then I need to substract game memory offset.

**Comment by slyyyy78** [comment]
Ok. Now I understand. I need  to find pointer's address.

**Comment by VladimirLuzhin** [comment]
Has anyone managed to find the required address yet?

**Comment by slyyyy78** [comment]
Ok , now I think I finally found it. 
5 static pointers always points to deaths counter even if I will restart game, chose new game etc.

But i have one more question where to find process address. Is it "eldenring.exe" ? 

So, choosing one pointer (but I'm not sure that I should add offset ?)
I think the only change which w need to introduce is:

`            
new Game("Sekiro", new int[] {0x3D5AAC0, 0x90},
new Game("eldenring", new int[] {0x3A42AA8, 0x90},
)
`

<?xml version="1.0" encoding="utf-8"?>
<CheatTable>
  <CheatEntries>
    <CheatEntry>
      <ID>2</ID>
      <Description>"pointerscan result"</Description>
      <VariableType>4 Bytes</VariableType>
      <Address>"eldenring.exe"+03A42AA8</Address>
      <Offsets>
        <Offset>20C</Offset>
      </Offsets>
    </CheatEntry>
    <CheatEntry>
      <ID>3</ID>
      <Description>"pointerscan result"</Description>
      <VariableType>4 Bytes</VariableType>
      <Address>"eldenring.exe"+03A42AB0</Address>
      <Offsets>
        <Offset>20C</Offset>
      </Offsets>
    </CheatEntry>
    <CheatEntry>
      <ID>4</ID>
      <Description>"pointerscan result"</Description>
      <VariableType>4 Bytes</VariableType>
      <Address>"eldenring.exe"+03C481D8</Address>
      <Offsets>
        <Offset>94</Offset>
      </Offsets>
    </CheatEntry>
    <CheatEntry>
      <ID>5</ID>
      <Description>"pointerscan result"</Description>
      <VariableType>4 Bytes</VariableType>
      <Address>"eldenring.exe"+03C65398</Address>
      <Offsets>
        <Offset>1F4</Offset>
      </Offsets>
    </CheatEntry>
    <CheatEntry>
      <ID>6</ID>
      <Description>"pointerscan result"</Description>
      <VariableType>4 Bytes</VariableType>
      <Address>"eldenring.exe"+045253F0</Address>
      <Offsets>
        <Offset>1F4</Offset>
      </Offsets>
    </CheatEntry>
  </CheatEntries>
</CheatTable>

**Comment by IgorGHK** [comment]
guys, you are awesome!
waiting for patch

**Comment by Quidrex** [comment]
https://www.reddit.com/r/ultrawidemasterrace/comments/t0xwov/flawless_widescreen_has_the_elden_ring_fix_out_now/

TL;DR from the comments: You're running into serious risk of a ban by triggering EAC or leaving it disabled for prolonged periods of time.
As such this death counter will probably NOT be updated for Elden Ring, at least not by reading process memory as I really don't want people to be banned by using it unless we can be absolutely sure that PROCESS_VM_READ does not trigger EAC.

I'm looking into watching the savefile for changes and reading it from there as a much less intrusive solution.

**Comment by slyyyy78** [comment]
No, My way to find correct addres is disabling internet connection. If I have internet connection and installed cheatengine then game/EAC doesn't allow to run.

**Comment by Quidrex** [comment]
@slyyyy78 Thank you for your research. Here is a test build using the first address from your search, [0x3A42AA8]+0x20.

USE AT YOUR OWN RISK. WE DO NOT KNOW YET IF EAC CAN OR WILL DETECT THIS.

[DSDeaths.zip](https://github.com/Quidrex/DSDeaths/files/8146656/DSDeaths.zip)

**Comment by Quidrex** [comment]
Does it really detect Cheat Engine being installed without even running? That's a level of intrusive software I do not tolerate on my computer.

**Comment by slyyyy78** [comment]
Yes, It detects and what worse I uninstalled Cheat Engine and it's still detects because drivers signing is disabled. 
I gave up . Sometime the counter works sometimes not.

**Comment by Quidrex** [comment]
I have ideas how to get it the information from the savefile without EAC caring. At least I hope EAC won't care. We can read the save file using opportunistic locking and just abort the try if Elden Ring tries so write to it.

**Comment by Danzoui** [comment]
I found the address - [0x3C481D8]+0x94
Works great for me. Try:
**USE AT YOUR OWN RISK**
[DSDeaths.zip](https://github.com/Quidrex/DSDeaths/files/8147129/DSDeaths.zip)

**Comment by Whalee110** [comment]
> @slyyyy78 Thank you for your research. Here is a test build using the first address from your search, [0x3A42AA8]+0x20.
> 
> USE AT YOUR OWN RISK. WE DO NOT KNOW YET IF EAC CAN OR WILL DETECT THIS.
> 
> [DSDeaths.zip](https://github.com/Quidrex/DSDeaths/files/8146656/DSDeaths.zip)

Works for me! - I am running with EAC disabled - prior though I had no issues runnning despite cheat engine being installed so not sure the issue there, perhaps some remenants of CE kept running?

**Comment by alterNERDtive** [comment]
So I guess this 
![image](https://user-images.githubusercontent.com/53827887/155879399-3919fef8-36ef-4631-bd62-11f57486fba0.png)
means “no can do with EAC on”? m)

(And how would I even disable it?)

**Comment by alterNERDtive** [comment]
Eh, I’d take the “small chance of race condition”. Worst case? Game crashes and you lose some progress.

**Comment by Whalee110** [comment]
To launch without EAC create steam_appid.txt in game directory and put int he value "1245620" then launch the game from the exe in the game directory. Obvs offline is disabled.

**Comment by Danzoui** [comment]
@Quidrex I think you can make some kind of warning / confirmation in the program itself for those who will use it in Elden Ring.

**Comment by VladimirLuzhin** [comment]
@Quidrex did you manage to read the save file?

**Comment by Quidrex** [comment]
I'll close this for now, a version reading from memory has just been released. Thanks to everyone for their research. I'll create a new ticket for switching over to save file reading depending on what the next days bring regarding patches and EAC.
